src/ralph/models/event_validator.py

```diff
@@ -0,0 +1,75 @@
+"""Event validator definition"""
+
+
+import json
+import logging
+from typing import TextIO
+
+from pydantic.error_wrappers import ValidationError
+
+from ralph.exceptions import BadFormatException, UnknownEventException
+from ralph.models.selector import ModelSelector
+
+logger = logging.getLogger(__name__)
+
+
+class EventValidator:
+    """Validates events using pydantic models."""
+
+    def __init__(self, model_selector: ModelSelector):
+        """Initialize EventValidator"""
+
+        self.model_selector = model_selector
+
+    def validate(self, input_file: TextIO, ignore_errors: bool, fail_on_unknown: bool):
+        """Validates JSON event strings line by line."""
+
+        count = 0
+        success_count = 0
+        for event_str in input_file:
+            try:
+                count += 1
+                yield self._validate_event(event_str)
+                success_count += 1
+            except (json.JSONDecodeError, TypeError) as err:
+                message = "Input event is not a valid JSON string"
+                self._log_error(message, event_str, err)
+                if not ignore_errors:
+                    raise BadFormatException(message) from err
+            except UnknownEventException as err:
+                if fail_on_unknown:
+                    raise err
+            except ValidationError as err:
+                message = f"Input event is not a valid {err.model.__name__}"
+                self._log_error(message, event_str, err)
+                if not ignore_errors:
+                    raise BadFormatException(message) from err
+        logger.info(
+            "Total events: %d, Invalid events: %d", count, count - success_count
+        )
+
+    def _validate_event(self, event_str: str):
+        """Validates a single JSON string event.
+
+        Raises:
+            TypeError: when the event_str is not of type string.
+            JSONDecodeError: when the event_str is not a valid JSON string.
+            UnknownEventException: when no matching model is found for the event.
+            ValidationError: when the event is failing the pydantic model validation.
+        Returns:
+            event_str (str): The unchanged input event_str.
+        """
+
+        event = json.loads(event_str)
+        model = self.model_selector.get_model(event)
+        if not model:
+            message = "No matching pydantic model found for input event."
+            self._log_error(message, event_str)
```

Comment:
Don't you think this error should be raised by the `ModelSelector.get_model()` method?

Reply:
Good Idea) Thanks) let's delegate this part to `ModelSelector`)